controllers/server/virtualapp_controller.go

```diff
@@ -186,7 +186,10 @@ func (r *VirtualAppReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 					klog.Infof("VApp %s/%s find Pod %s subset limit resourcesMatching changed, going to reload it.", vApp.Namespace, vApp.Name, pod.Name)
 					diffState.needReload = true
 				}
-				// TODO: if objectSelector changed, we might have to reload all?
+				if newLimit.ObjectSelector != currentLimit.ObjectSelector {
+					klog.Infof("VApp %s/%s find Pod %s subset limit objectSelector changed, going to reload it.", vApp.Namespace, vApp.Name, pod.Name)
+					diffState.needReload = true
```

Comment:
Should we reload all subsets? For some other subsets may have already watched and handled the objects that changed by selector.